src/Core/System/SystemConfig/AbstractSystemConfigLoader.php

```diff
@@ -9,5 +9,8 @@ abstract class AbstractSystemConfigLoader
 {
     abstract public function getDecorated(): AbstractSystemConfigLoader;
 
+    /**
+     * @return array<mixed>
```

Comment:
the system config is always an associative array, so 
```suggestion
     * @return array<string, mixed>
```

please change all other config related stuff accordingly :slightly_smiling_face: